File: docs/src/testers/sessions/lifecycle/monitor_state.ts
// @sniptest filename=monitor_state.ts
// @sniptest show=6-12
import { NotteClient, type SessionResponse } from 'notte-sdk';

const client = new NotteClient();
let status: SessionResponse | undefined;

const session = client.Session();
await session.use(async () => {
  status = await session.status();
  if (status.status !== 'active') throw new Error('Session is not active');
  console.log(`Status: ${status.status}`);
});
// Automatically stopped here.

export { status };

File: docs/src/testers/sessions/lifecycle/monitor_state.py
# @sniptest filename=monitor_state.py
import json

from notte_sdk import NotteClient

client = NotteClient()

with client.Session(idle_timeout_minutes=2) as session:
    status = session.status()
    if status.status != "active":
        raise RuntimeError("Session is not active")
    print(
        json.dumps(
            {
                "session_id": session.session_id,
                "status": status.status,
                "idle_timeout_minutes": status.idle_timeout_minutes,
            }
        )
    )
# Automatically stopped here.
